iceoryx_utils/test/moduletests/test_posix_mutex.cpp

```diff
@@ -26,15 +28,6 @@ using namespace iox::units::duration_literals;
 class Mutex_test : public Test
 {
   public:
```

Comment:
I think all of the unit tests can have the potential to deadlock when our mutex implementation is flawed. In this case it would be ideal that the unit tests itself discovers the deadlock and terminates the whole tests so that it is not running forever.
@FerdinandSpitzschnueffler wrote the `iceoryx_utils/testutils/watch_dog.hpp` class, could you please use it here like
```cpp
class Mutex_test : public Test {
   // ...
   iox::units::Duration m_watchdogTimeout = 2_s;
   Watchdog m_deadlockWatchdog(m_watchdogTimeout);
   void SetUp() override {
      // ...
      m_deadlockWatchdog.watchAndActOnFailure([]{ std::terminate(); });
   }
};
```